### PR TITLE
Grubenv

### PR DESCRIPTION
**Pull request by omgold**
Created a lense for grubenv (grub2)

### CODE REVIEW

**Comment by raphink** [comment]
LGTM. Might be good to add NaturalDocs documentation, too. See `keepalived.aug` and `test_keepalived.aug` for examples.

File: lenses/grubenv.aug [hunk 1]
```diff
@@ -0,0 +1,19 @@
+(* Parsing /boot/grub/grubenv *)
+
+module GrubEnv =
+  autoload xfm
+
+  let eol = Util.del_str "\n"
```
Comment (raphink) [inline lenses/grubenv.aug]: Please use `Util.eol` like all other lenses.

File: lenses/grubenv.aug [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  let comment = Util.comment
+  let eq = Util.del_str "="
```
Comment (raphink) [inline lenses/grubenv.aug]: Use `Sep.equal` for that.

File: lenses/grubenv.aug [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  let value   = /[^\\\n]*(\\\\(\\\\|\n)[^\\\n]*)*/
```
Comment (raphink) [inline lenses/grubenv.aug]: `Rx.cl` might help here.

File: lenses/grubenv.aug [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  let word = /[A-Za-z_][A-Za-z0-9_]*/
```
Comment (raphink) [inline lenses/grubenv.aug]: If you're not too strict on possible values, `Rx.word` would work here.

**Comment by raphink** [comment]
@omgold any news on this PR?

**Comment by lutter** [comment]
@raphink can we merge this as-is ? Or do you want to touch up the changes you suggested and then merge that ?

**Pull request merged**